File: Java/BOJ_2750_selection.java
/*
 *  Sorting numbers
 *  Using Selection Sort
 *
 *  https://www.acmicpc.net/problem/2750
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

class BOJ_2750 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int [] data = new int [N];
        for (int i = 0; i < N; i++) {
            data[i] = Integer.parseInt(br.readLine());
        }
        BOJ_2750 main = new BOJ_2750();
        main.selectionSort(data);
        for (int i : data) {
            System.out.println(i);
        }
    }
    void selectionSort (int [] arr) {
        int indexMin, temp;
        for (int i = 0; i < arr.length; i++) {
            indexMin = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[indexMin]) {
                    indexMin = j;
                }
            }
            if (indexMin != i) {
                temp = arr[i];
                arr[i] = arr[indexMin];
                arr[indexMin] = temp;
            }
        }
    }
}

File: Python/BOJ_2750_selection.py
"""
Sorting numbers
Using Selection Sort

https://www.acmicpc.net/problem/2750
"""


def sol():
    data = []
    for _ in range(int(input())):
        data.append(int(input()))
    selection(data)
    return data


def selection(data):
    for index in range(len(data)):
        index_min = index
        for compare_index in range(index, len(data)):
            if data[index_min] > data[compare_index]:
                index_min = compare_index
        if index_min != index:
            temp = data[index]
            data[index] = data[index_min]
            data[index_min] = temp


if __name__ == "__main__":
    for i in sol():
        print(i)
